Frameworks/CoreGraphics/DWriteFontCollectionHelper.mm

```diff
@@ -0,0 +1,207 @@
+    uint32_t count = fontCollection->GetFontFamilyCount();
+    CFMutableArrayRef ret = CFArrayCreateMutable(kCFAllocatorSystemDefault, count, &kCFTypeArrayCallBacks);
+    for (uint32_t i = 0; i < count; ++i) {
+        // Get the font family.
+        ComPtr<IDWriteFontFamily> fontFamily;
+        RETURN_NULL_IF_FAILED(fontCollection->GetFontFamily(i, &fontFamily));
+
+        // Get a list of localized strings for the family name.
+        ComPtr<IDWriteLocalizedStrings> familyNames;
+        RETURN_NULL_IF_FAILED(fontFamily->GetFamilyNames(&familyNames));
+
+        CFStringRef name = _CFStringFromLocalizedString(familyNames.Get());
+        if (CFStringGetLength(name) == 0) {
+            TraceError(TAG, L"Failed to convert the localized string to wide string.");
+            continue;
+        }
+
+        CFArrayAppendValue(ret, name);
+    }
+
+    return ret;
+}
+
+CFMutableArrayRef DWriteFontCollectionHelper::CopyFontNamesForFamilyName(CFStringRef familyName) {
+    CFMutableArrayRef ret = CFArrayCreateMutable(kCFAllocatorSystemDefault, 0, &kCFTypeArrayCallBacks);
+
+    ComPtr<IDWriteFontList> fontList;
+    if (FAILED(_GetFontListForFamilyName(familyName, &fontList))) {
+        return ret;
+    }
+
+    size_t fontCount = fontList->GetFontCount();
+    for (size_t i = 0; i < fontCount; i++) {
+        ComPtr<IDWriteFont> font;
+        if (FAILED(fontList->GetFont(i, &font))) {
+            continue;
+        }
+
+        ComPtr<IDWriteLocalizedStrings> fullName;
+        BOOL exist = FALSE;
+        if (FAILED(font->GetInformationalStrings(DWRITE_INFORMATIONAL_STRING_FULL_NAME, &fullName, &exist))) {
+            continue;
+        }
+
+        if (exist) {
+            CFStringRef name = _CFStringFromLocalizedString(fullName.Get());
+            if (CFStringGetLength(name) == 0) {
+                TraceError(TAG, L"Failed to convert the localized string to wide string.");
+                continue;
+            }
+
+            CFArrayAppendValue(ret, name);
+        }
+    }
+
+    return ret;
+}
+
+std::shared_ptr<_DWriteFontProperties> DWriteFontCollectionHelper::GetFontPropertiesFromUppercaseFontName(
+    const woc::unique_cf<CFStringRef>& upperFontName) {
+    if (!m_propertiesMap) {
+        _InitializePropertiesMap();
+    }
+
+    const auto& info = m_propertiesMap->find(upperFontName);
+    if (info != m_propertiesMap->end()) {
+        return info->second;
+    }
+
+    return nullptr;
+}
+
+HRESULT DWriteFontCollectionHelper::CreateFontFamilyWithName(_In_ const wchar_t* unicharFamilyName,
+                                                             _Outptr_ IDWriteFontFamily** outFontFamily) {
+    ComPtr<IDWriteFontCollection> fontCollection = _GetFontCollection();
+
+    // If no/empty font collection, say that the font family couldn't be found
+    // Use S_FALSE to represent 'not found'
+    RETURN_HR_IF_NULL(S_FALSE, fontCollection);
+
+    size_t fontFamilyIndex;
+    BOOL fontFamilyExists;
```

Comment:
same as the other uninitialized comment ([https://msdn.microsoft.com/en-us/library/windows/desktop/dd368217(v=vs.85).aspx](https://msdn.microsoft.com/en-us/library/windows/desktop/dd368217(v=vs.85).aspx))

---
In reply to: [90742919](https://github.com/Microsoft/WinObjC/pull/1504#discussion_r90742919) [](ancestors = 90742919)